### ISSUE TITLE
koverGenerateArtifact task generates a cacheable report that contains absolute path

### ISSUE THREAD

**Issue by quanturium** [Kotlin/kotlinx-kover#440]
**Describe the bug**
`koverGenerateArtifact` task generates a report that contains absolute paths. That task is cacheable which means that subsequent builds pulling from the cache will get the same task output. The problem is that absolute paths are not the same on different machines, leading to subsequent tasks (e.g. koverXmlReport) to process invalid paths on subsequent builds/machines and as a result generates an empty report

**Expected behavior**
`koverGenerateArtifact` should generate the content of its output as relative paths to the root of the project. This should be an easy fix. 

**Reports**
Example of the output file from koverGenerateArtifact stored in our build cache:
```
/var/lib/buildkite-agent/builds/xxx/android-diff/build-src/lib-android-lint/src/main/kotlin
/var/lib/buildkite-agent/builds/xxx/android-diff/build-src/lib-android-lint/src/main/java

/var/lib/buildkite-agent/builds/xxx/android-diff/build-src/lib-android-lint/build/classes/java/main
/var/lib/buildkite-agent/builds/xxx/android-diff/build-src/lib-android-lint/build/classes/kotlin/main

/var/lib/buildkite-agent/builds/xxx/android-diff/build-src/lib-android-lint/build/kover/bin-reports/test.ic
```

**Comment by quanturium** [comment]
Also, here is a workaround in the meantime for anyone facing the same issue: 

```
project.tasks.named("koverGenerateArtifact") {
            outputs.cacheIf { false }
        }
```

**Comment by quanturium** [comment]
One more thought on this, @shanshin do you think this task should be cacheable? It seems like it's a really fast one generating a small file that might be faster to be run than to be pulled from a remote cache.

**Comment by shanshin** [comment]
I have already prepared a fix. In any case, using absolute paths was originally a bad idea.

> It seems like it's a really fast one generating a small file that might be faster to be run than to be pulled from a remote cache

I'm not sure that the difference will be significant, but if you try it on yourself, it will be useful. 

Since we do not have statistics on the use of the remote cache, therefore, by default we assume that the local cache is used, and for special cases, it is possible to  disable caching.

**Comment by quanturium** [comment]
>I have already prepared a fix. In any case, using absolute paths was originally a bad idea.
Awesome! Thank you. When can we expect a fix for this one? 0.7.4?

> I'm not sure that the difference will be significant, but if you try it on yourself, it will be useful.
I can give you data on this once we have a fix (0.7.4). I took a quick look at what this task does and it looks like it's only writing a few strings (file paths) to a file. I can guarantee you this is not worth caching due to the extremely low amount of computation. Even copy tasks are not considered worth caching according to Gradle: https://docs.gradle.org/current/userguide/build_cache.html#sec:task_output_caching_disabled_by_default

Note that not making it cacheable doesn't mean subsequent tasks (tasks depending on `koverGenerateArtifact`)  won't be cached (like koverXmlReport) as long as the inputs are the same.

**Comment by shanshin** [comment]
Optimistically, `0.7.4` should be released next week
